lib/logstash/filters/aggregate.rb

```diff
@@ -202,13 +216,20 @@ def filter(event)
 			map = aggregate_maps_element.map
 
 			# execute the code to read/update map and event
-			@codeblock.call(event, map)
+			begin
+				@codeblock.call(event, map)
+				noError = true
+			rescue => exception
+				@logger.error("Aggregate exception occurred. Error: #{exception} ; Code: #{@code} ; Map: #{map} ; EventData: #{event.instance_variable_get('@data')}")
+				event.tag("_aggregateexception")
+			end
 			
 			# delete the map if task is ended
 			@@aggregate_maps.delete(task_id) if @end_of_task
 		end
 
-		filter_matched(event)
+		# match the filter, only if no error occurred
+		filter_matched(event) if noError
```

Comment:
@purbon 
"noError" is a local variable in method "filter".
So there is no synchronization issue with this variable.